kratos/spatial_containers/geometrical_objects_bins.cpp

```diff
@@ -89,16 +96,35 @@ void GeometricalObjectsBins::SearchInRadius(
         for(std::size_t j = min_position[1]; j < max_position[1]; j++) {
             for(std::size_t i = min_position[0]; i < max_position[0]; i++) {
                 auto& r_cell = GetCell(i, j, k);
-                for(auto p_geometrical_object : r_cell) {
-                    candidates.insert(p_geometrical_object);
+                if (IsCellBoundingBoxInsideRadius(i, j, k, rPoint, Radius)) {
```

Comment:
I would check if the cell is empty before this if to avoid the unnecessary computation of cell is inside